api/src/routes/email-subscription.ts

```diff
@@ -0,0 +1,136 @@
+/* eslint-disable @typescript-eslint/naming-convention */
+import { type FastifyPluginCallbackTypebox } from '@fastify/type-provider-typebox';
+import { schemas } from '../schemas';
+import { getRedirectParams } from '../utils/redirection';
+
+/**
+ * Endpoints to set 'sendQuincyEmail' to true or false using 'unsubscribeId'.
+ *
+ * @param fastify The Fastify instance.
+ * @param _options Options passed to the plugin via `fastify.register(plugin, options)`.
+ * @param done The callback to signal that the plugin is ready.
+ */
+export const emailSubscribtionRoutes: FastifyPluginCallbackTypebox = (
+  fastify,
+  _options,
+  done
+) => {
+  fastify.get(
+    '/ue/:unsubscribeId',
+    {
+      schema: schemas.unsubscribe,
+      errorHandler(error, request, reply) {
+        if (error.validation) {
+          const { origin } = getRedirectParams(request);
+          void reply.code(302);
+          void reply.redirectWithMessage(origin, {
+            type: 'info',
+            content: 'We could not find an account to unsubscribe.'
+          });
+        } else {
+          fastify.errorHandler(error, request, reply);
+        }
+      }
+    },
+    async (req, reply) => {
+      try {
+        const { origin } = getRedirectParams(req);
+        const { unsubscribeId } = req.params;
+        const users = await fastify.prisma.user.findMany({
+          where: { unsubscribeId }
+        });
+
+        if (!users.length) {
+          void reply.code(302);
+          return reply.redirectWithMessage(origin, {
+            type: 'info',
+            content: 'We could not find an account to unsubscribe.'
+          });
+        }
+
+        for (const user of users) {
+          await fastify.prisma.user.update({
+            where: { id: user.id },
+            data: {
+              sendQuincyEmail: false
+            }
+          });
+        }
+
+        return reply.redirectWithMessage(
+          `${origin}/unsubscribed/${unsubscribeId}`,
+          {
+            type: 'success',
+            content: "We've successfully updated your email preferences."
+          }
+        );
+      } catch (error) {
+        fastify.log.error(error);
+        void reply.code(302);
+        return reply.redirectWithMessage(origin, {
+          type: 'danger',
+          content: 'Something went wrong.'
+        });
+      }
+    }
+  );
+
+  fastify.get(
+    '/resubscribe/:unsubscribeId',
+    {
+      schema: schemas.resubscribe,
+      errorHandler(error, request, reply) {
+        if (error.validation) {
+          const { origin } = getRedirectParams(request);
+          void reply.code(302);
+          void reply.redirectWithMessage(origin, {
+            type: 'info',
+            content:
+              'We were unable to process this request, please check and try again.'
```

Comment:
The original string is:

> We we unable to process this request, please check and try again

We we not be fixing bugs until v1.0 of the new api.

Reply:
really?

Reply:
The heuristic I've been using is "Does the client or mobile rely on the thing? Don't change it.". This doesn't seem to be the case here.